drivers/crypto/crypto_rts5912_sha.c

```diff
@@ -0,0 +1,278 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Copyright (c) 2025 Realtek Semiconductor Corporation, SIBG-SD7
+ *
+ */
+
+#define DT_DRV_COMPAT realtek_rts5912_sha
+
+#include <errno.h>
+#include <string.h>
+#include <zephyr/init.h>
+#include <zephyr/crypto/crypto.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(sha256_hw_rtk, CONFIG_CRYPTO_LOG_LEVEL);
+
+BUILD_ASSERT(DT_NUM_INST_STATUS_OKAY(DT_DRV_COMPAT) == 1,
+	     "only one realtek,rts5912-sha compatible node can be supported");
+
+#include "reg/reg_crypto.h"
+#include "crypto_rts5912_priv.h"
+
+static struct rts5912_sha256_context rts5912_sha256_ctx;
+
+#define RTS5912_SHA2DMA_MAXIMUM_BLOCK_NUM    (0x1FFUL)
+#define RTS5912_SHA2DMA_BLOCK_SIZE           (64)
+#define RTS5912_SHA2DMA_BLOCK_SHIFT          (6)
+#define RTS5912_SHA2DMA_8Byte_SHIFT          (3)
+#define RTS5912_SHA2DMA_DST_WIDTH            (0x3)
+#define RTS5912_SHA2DMA_SRC_WIDTH            (0x3)
+#define RTS5912_SHA2DMA_HIGH_LEVEL_MSK       (0xFFFF0000)
+#define RTS5912_SHA2_BLOCK_EXTEND_CHECK      (56)
+#define RTS5912_MAXIMUM_CRYPTO_POLLING_TIMES (50)
```

Comment:
Include the units in the polling time macro
```suggestion
#define RTS5912_MAXIMUM_CRYPTO_POLLING_TIME_US (50 * USEC_PER_MSEC)
```